### PR TITLE
kubekey 2.3.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  # github.com/proglottis/gpgme
  /github/home/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/proglottis/gpgme@v0.1.1/data.go:4:11: fatal error: gpgme.h: No such file or directory
      4 | // #include <gpgme.h>
        |           ^~~~~~~~~
  compilation terminated.
  # pkg-config --cflags  -- devmapper
  pkg-config: exec: "pkg-config": executable file not found in $PATH
  # github.com/containers/storage/drivers/btrfs
  /github/home/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/containers/storage@v1.40.0/drivers/btrfs/btrfs.go:9:10: fatal error: btrfs/ioctl.h: No such file or directory
      9 | #include <btrfs/ioctl.h>
        |          ^~~~~~~~~~~~~~~
  compilation terminated.
  
```

**Comment by SMillerDev** [comment]
```
  /github/home/.cache/Homebrew/go_mod_cache/pkg/mod/github.com/containers/storage@v1.40.0/drivers/btrfs/btrfs.go:9:10: fatal error: btrfs/ioctl.h: No such file or directory
      9 | #include <btrfs/ioctl.h>
        |          ^~~~~~~~~~~~~~~
```

**Comment by chenrui333** [comment]
Gonna make the build macos only (similar build issue with werf)

File: Formula/kubekey.rb [hunk 1]
```diff
@@ -16,6 +17,9 @@ class Kubekey < Formula
   end
 
   depends_on "go" => :build
+  depends_on "gpgme"
+  # due to missing libbtrfs headers, only supports macos at the moment
```
Comment (SMillerDev) [inline Formula/kubekey.rb:21]: Any issue upstream for this?

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/kubekey.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/kubekey.rb:21]: This is basically the same thing I did with werf

https://github.com/Homebrew/homebrew-core/blob/edbc4965f09e9f82322464d52dd34bde46d50237/Formula/werf.rb#L26-L27

Comment (chenrui333) [reply Formula/kubekey.rb:21]: We basically just missed the libbtrfs formula for the linux build, there is nothing wrong with the upstream.

Comment (SMillerDev) [reply Formula/kubekey.rb:21]:
```suggestion
  # due to missing libbtrfs formula, not needed on macOS
```

Comment (cho-m) [reply Formula/kubekey.rb:21]: May be available once we add `btrfs-progs` (#114864)

**Comment by chenrui333** [comment]
🎉 this is good now, thanks @cho-m!

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**